third_party/nvfuser/python_tests/test_torchscript.py

```diff
@@ -4201,6 +4201,25 @@ def t(x: torch.Tensor, y: torch.Tensor, ind: torch.Tensor):
         t_jit = torch.jit.script(t)
         self._run_training_helper(t_jit, t, grad, x, y, ind)
 
+    @unittest.skipIf(not RUN_NVFUSER, "requires CUDA")
+    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING,
+                     "Requires fusion optimization pass to be effective")
+    def test_index_select_shape_expression(self):
+        x = torch.randn([68, 128], dtype=torch.float, device="cuda")
+        y = torch.randn_like(x)
+        ind = torch.randint(0, 68, (130,), device="cuda").to(dtype=torch.int)
+
+        def t(x: torch.Tensor, y: torch.Tensor, ind: torch.Tensor):
+            o = torch.mul(x, y)
+            o = torch.index_select(o, 0, ind)
+            return o.size()
+
+        t_jit = torch.jit.script(t)
+        jit_o = t_jit(x, y, ind)
+        jit_o = t_jit(x, y, ind)
```

Comment:
Looks good. Can we also assert that we actually have the the short-cut for shape in-place?
something like this vvv.
https://github.com/csarofeen/pytorch/blob/75ce4aeb896c7bc0558e677f2b548f0312f791ae/third_party/nvfuser/python_tests/test_torchscript.py#L3929-L3930

Reply:
Thanks for your tips! Done.